sklearn/linear_model/ridge.py

```diff
@@ -326,11 +348,31 @@ def ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
     -----
     This function won't compute the intercept.
     """
-    if return_intercept and sparse.issparse(X) and solver != 'sag':
+
+    return _ridge_regression(X, y, alpha,
+                             sample_weight=sample_weight,
+                             solver=solver,
+                             max_iter=max_iter,
+                             tol=tol,
+                             verbose=verbose,
+                             random_state=random_state,
+                             return_n_iter=return_n_iter,
+                             return_intercept=return_intercept,
+                             X_scale=None,
+                             X_offset=None)
+
+
+def _ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
+                     max_iter=None, tol=1e-3, verbose=0, random_state=None,
+                     return_n_iter=False, return_intercept=False,
+                     X_scale=None, X_offset=None):
+
+    if (return_intercept and sparse.issparse(X) and
+       solver not in ['sag', 'sparse_cg']):
         if solver != 'auto':
-            warnings.warn("In Ridge, only 'sag' solver can currently fit the "
-                          "intercept when X is sparse. Solver has been "
-                          "automatically changed into 'sag'.")
+            warnings.warn("In Ridge, only 'sag' and 'sparse_cg' solvers "
+                          "can currently fit the intercept when X is sparse."
+                          " Solver has been automatically changed into 'sag'.")
```

Comment:
this is not really true as you cannot pass return_intercept=True in ridge_regression with sparse_cg. It does the job with X_offset and X_scale.

Reply:
I reversed this change.